tests/python/relax/test_bind_symbolic_vars.py

```diff
@@ -183,6 +183,91 @@ def expected(A: R.Tensor(["M", 16])):
     tvm.ir.assert_structural_equal(expected, after)
 
 
+def test_bind_symbolic_vars_in_shape_expr():
+    """The bound variable should be replaced when appearing in R.Shape"""
+
+    @R.function(private=True)
+    def before(A: R.Tensor(["M * N"]), x: R.Shape(["M", "N"])):
```

Comment:
Are we sure we want to permit cases like this? I believe I wrote the specification draft to require argument vars to be in binding positions, left to right. If we permit uses before the binding position, then should the type-checker scan through the arguments, identify the bindings first, and only then check the position?

Reply:
I was thinking so, mainly because it would be the same support that is provided in TIR.  (`ArgBinder::BindDLTensor` defines the symbolic variables prior to asserting based on the shapes.)

There's also a couple of edge cases that I think would be easier to handle by allowing it.  Swapping the order of commutative operators, hoisting `R.match_cast` from the body into the function signature, and handling cases with mutually-dependent shapes.  (e.g. The first parameter is `R.Tensor(["a * b", "b"])` and the second parameter is `R.Tensor(["a * b", "a"])`.)

Reply:
Okay, I'll have to modify the specification to account for it.

Reply:
Sounds good, and thank you for all your work keeping the spec up to date!